core/src/main/java/hudson/PluginManager.java

```diff
@@ -137,7 +137,25 @@
     /**
      * All active plugins, topologically sorted so that when X depends on Y, Y appears in the list before X does.
      */
-    protected final List<PluginWrapper> activePlugins = new CopyOnWriteArrayList<PluginWrapper>();
+    protected final List<PluginWrapper> activePlugins = new CopyOnWriteArrayList<PluginWrapper>() {
+        @Override
+        public boolean add(PluginWrapper pluginWrapper) {
+            try {
+                return super.add(pluginWrapper);
+            } finally {
+                onActivate(pluginWrapper);
```

Comment:
It makes sense to let listener know about the result or an exception if it happens